src/merkle/smt/full/mod.rs

```diff
@@ -172,6 +167,47 @@ impl Smt {
         <Self as SparseMerkleTree<SMT_DEPTH>>::insert(self, key, value)
     }
 
+    /// Compute what changes are necessary to insert the specified key-value pairs into this Merkle
+    /// tree, allowing for validation before applying those changes.
+    ///
+    /// This method returns a [`MutationSet`], which contains all the information for inserting
+    /// `kv_pairs` into this Merkle tree already calculated, including the new root hash, which can
+    /// be queried with [`MutationSet::root()`]. When you are done, call [`Smt::apply_mutations()`]
```

Comment:
I would write: Once a mutation set is returned,` [Smt::apply_mutations()]` can be called in order to commit ...